bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java

```diff
@@ -44,45 +47,93 @@
 import io.github.hapjava.accessories.HomekitAccessory;
 import io.github.hapjava.server.impl.HomekitRoot;
 import io.github.hapjava.server.impl.HomekitServer;
+import io.github.hapjava.server.impl.crypto.HAPSetupCodeUtils;
 
 /**
  * Provides access to openHAB items via the HomeKit API
  *
  * @author Andy Lintner - Initial contribution
  */
-@Component(service = { Homekit.class }, configurationPid = "org.openhab.homekit", property = {
+@Component(service = { Homekit.class }, configurationPid = HomekitSettings.CONFIG_PID, property = {
         Constants.SERVICE_PID + "=org.openhab.homekit", "port:Integer=9123" })
 @ConfigurableService(category = "io", label = "HomeKit Integration", description_uri = "io:homekit")
 @NonNullByDefault
 public class HomekitImpl implements Homekit {
     private final Logger logger = LoggerFactory.getLogger(HomekitImpl.class);
-    private final NetworkAddressService networkAddressService;
-    private final HomekitChangeListener changeListener;
 
-    private HomekitSettings settings;
+    @Reference
```

Comment:
Why did you change back from constructor injection to field injection?

Reply:
i have followed the example from https://www.openhab.org/docs/developer/osgi/configadmin.html and from hueemulation..ConfigStore 
are there any recommendation?

Reply:
That needs to be updated. Thanks. The recommended way is to use constructor injection, like it was before. You can just add the `ConfigurationAdmin` as additionl parameter  to the constructor (`@Reference ConfigurationAdmin configAdmin`) and OSGi will make sure it is available before the `HomekitImpl` service starts.

Reply:
done